index.js

```diff
@@ -145,16 +145,34 @@ function $asInteger (i) {
 }
 
 function $asNumber (i) {
```

Comment:
generically, I _prefer_ if we firstly check if `i` is valid and return that, and then handle everything else.

Reply:
I'll do my best

Reply:
done